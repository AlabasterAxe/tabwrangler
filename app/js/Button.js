import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default class Button extends React.PureComponent {
  constructor() {
    super();
  }

  render() {
    const classes = classNames(['glyphicon', this.props.className]);
    const content = <span><i className={classes}></i> {this.props.label}</span>;

    return (
      <button className="btn btn-default btn-xs" onClick={this.props.clickHandler}>
        {content}
      </button>
    );
  }
}

Button.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};