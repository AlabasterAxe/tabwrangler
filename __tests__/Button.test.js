import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import Button from '../app/js/Button';

test('render button with correct label', () => {
  const mockCallback = jest.fn();

  const button = renderer.create(
    <Button label='Export' clickHandler={mockCallback} className='glyphicon-export'/>
  );

  expect(button).toMatchSnapshot();
});

test('should call click handler callback when clicked', () => {
  const mockCallback = jest.fn();

  const button = ReactTestUtils.renderIntoDocument(
    <Button label='Export' clickHandler={mockCallback} className='glyphicon-export'/>
  );
  const buttonNode = ReactTestUtils.findRenderedDOMComponentWithTag(button, 'button');
  ReactTestUtils.Simulate.click(buttonNode);

  expect(mockCallback.mock.calls.length).toBe(1);
});