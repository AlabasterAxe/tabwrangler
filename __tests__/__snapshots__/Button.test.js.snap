// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button with correct label 1`] = `
<button
  className="btn btn-default btn-xs"
  onClick={[Function]}
>
  <span>
    <i
      className=""
    />
    
    Export
  </span>
</button>
`;

exports[`render button with correct label and glyphicon 1`] = `
<button
  className="btn btn-default btn-xs"
  onClick={[Function]}
>
  <span>
    <i
      className="glyphicon glyphicon-export"
    />
     
    Export
  </span>
</button>
`;
